File: src/main/java/gargoyle/ct/prop/impl/CTSimpleEnumProperty.java
package gargoyle.ct.prop.impl;

public class CTSimpleEnumProperty<E extends Enum<E>> extends CTSimpleProperty<E> {
    public CTSimpleEnumProperty(Class<E> type, String name) {
        this(type, name, null);
    }

    public CTSimpleEnumProperty(Class<E> type, String name, E def) {
        super(name, def);
    }

    public CTSimpleEnumProperty(Class<E> type) {
        this(type, type.getSimpleName(), null);
    }

    public CTSimpleEnumProperty(Class<E> type, E def) {
        super(type.getSimpleName(), def);
    }

    public CTSimpleEnumProperty(String name, E def) {
        super(name, def);
    }

    public CTSimpleEnumProperty(E def) {
        super(def.getClass().getSimpleName(), def);
    }
}

File: util/src/main/kotlin/gargoyle/ct/util/prop/impl/simple/CTSimpleEnumProperty.kt
package gargoyle.ct.util.prop.impl.simple

import gargoyle.ct.util.util.simpleClassName
import kotlin.reflect.KClass

class CTSimpleEnumProperty<E : Enum<E>> : CTSimpleProperty<E> {
    constructor(type: KClass<E>, name: String) : super(type, name)
    constructor(type: KClass<E>) : super(type, type.simpleClassName)
    constructor(type: KClass<E>, def: E) : super(type.simpleClassName, def)
    constructor(name: String, def: E) : super(name, def)
    constructor(def: E) : super(def::class.simpleClassName, def)
}
